src/requests/requests.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@nestjs/common'
+import { PrismaService } from 'src/prisma/prisma.service'
+import { UserEntity } from 'src/users/dto/UserEntity.dto'
+import { CreateRequestDto } from './dto/CreateRequest.dto'
+import { RequestDetailsDto } from './dto/RequestDetails.dto'
+import { RequestPreviewDto } from './dto/RequestPreview.dto'
+import { UpdateRequestDto } from './dto/UpdateRequest.dto'
+
+@Injectable()
+export class RequestsService {
+  constructor(private prisma: PrismaService) {}
+
+  async findAll(): Promise<RequestPreviewDto[]> {
+    const requests = await this.prisma.consultationRequest.findMany({
+      include: {
+        initializer: true,
+        subject: true,
+        _count: {
+          select: {
+            supporters: true,
+            consultation: true,
+          },
+        },
+      },
+    })
+
+    return requests.map(({ initializer, _count, ...request }) => {
+      return {
+        ...request,
+        initializer: { fullName: initializer.fullName, id: initializer.id },
+        supporterCount: _count.supporters,
+        consultationCount: _count.consultation,
+      }
+    })
+  }
+
+  async findOne(id: number): Promise<RequestDetailsDto> {
+    const { initializer, supporters, consultation, ...request } =
+      await this.prisma.consultationRequest.findUnique({
+        where: {
+          id,
+        },
+        include: {
+          initializer: true,
+          subject: true,
+          supporters: true,
+          consultation: true,
+        },
+      })
```

Comment:
findUnique returns with null if the record isn't found. Not sure why the try-catch in the controller doesn't catch the exception thrown in this case, but it's probably nicer anyway if this case is handled in the service. So throw 404 here if the result is null.